src/components/views/settings/TchapJoinRuleSettings.tsx

```diff
@@ -0,0 +1,352 @@
+import { ViewRoomPayload } from "matrix-react-sdk/src/dispatcher/payloads/ViewRoomPayload";
+import { doesRoomVersionSupport, PreferredRoomVersions } from "matrix-react-sdk/src/utils/PreferredRoomVersions";
+import { UIComponent } from "matrix-react-sdk/src/settings/UIFeature";
+
+import { ComponentVisibilityCustomisations } from "../../../customisations/TchapComponentVisibility";
+
+ interface IProps {
+    room: Room;
+    promptUpgrade?: boolean;
+    closeSettingsFn(): void;
+    onError(error: Error): void;
+    beforeChange?(joinRule: JoinRule): Promise<boolean>; // if returns false then aborts the change
+    aliasWarning?: ReactNode;
+}
+
+const JoinRuleSettings = ({ room, promptUpgrade, aliasWarning, onError, beforeChange, closeSettingsFn }: IProps) => {
+    const cli = room.client;
+
+    const roomSupportsRestricted = doesRoomVersionSupport(room.getVersion(), PreferredRoomVersions.RestrictedRooms);
+    const preferredRestrictionVersion = !roomSupportsRestricted && promptUpgrade
+        ? PreferredRoomVersions.RestrictedRooms
+        : undefined;
+
+    const disabled = !room.currentState.mayClientSendStateEvent(EventType.RoomJoinRules, cli);
+
+    const [content, setContent] = useLocalEcho<IJoinRuleEventContent>(
+        () => room.currentState.getStateEvents(EventType.RoomJoinRules, "")?.getContent(),
+        content => cli.sendStateEvent(room.roomId, EventType.RoomJoinRules, content, ""),
+        onError,
+    );
+
+    const { join_rule: joinRule = JoinRule.Invite } = content || {};
+    const restrictedAllowRoomIds = joinRule === JoinRule.Restricted
+        ? content.allow?.filter(o => o.type === RestrictedAllowType.RoomMembership).map(o => o.room_id)
+        : undefined;
+
+    const editRestrictedRoomIds = async (): Promise<string[] | undefined> => {
+        let selected = restrictedAllowRoomIds;
+        if (!selected?.length && SpaceStore.instance.activeSpaceRoom) {
+            selected = [SpaceStore.instance.activeSpaceRoom.roomId];
+        }
+
+        const matrixClient = MatrixClientPeg.get();
+        const { finished } = Modal.createDialog(ManageRestrictedJoinRuleDialog, {
+            matrixClient,
+            room,
+            selected,
+        }, "mx_ManageRestrictedJoinRuleDialog_wrapper");
+
+        const [roomIds] = await finished;
+        return roomIds;
+    };
+
+    /* code from element web
+    const definitions: IDefinition<JoinRule>[] = [{
+        value: JoinRule.Invite,
+        label: _t("Private (invite only)"),
+        description: _t("Only invited people can join."),
+        checked: joinRule === JoinRule.Invite || (joinRule === JoinRule.Restricted && !restrictedAllowRoomIds?.length),
+    }, {
+        value: JoinRule.Public,
+        label: _t("Public"),
+        description: <>
+            { _t("Anyone can find and join.") }
+            { aliasWarning }
+        </>,
+    }]; */
+
+    //tchap : we do not permit to change the type of room, thus display only one option
```

Comment:
can you specify where tchap code ends, as well ? So that we know what's ours and what's not.